### PR TITLE
only set wrapper args for cross compilation, keeping subset of LDFLAGS necessary for sysroot < 2.17

### PR DESCRIPTION
**Pull request by minrk**
defaults are not likely to be wrong for native compilation, don't mess with them, I think #158 went a bit too far.

and don't passthrough $LDFLAGS, etc. from compiler activation, set the actual correct values from the wrappers with the right prefix.

closes #163

### CODE REVIEW

**Comment by minrk** [comment]
Still need to update the tests

File: recipe/build-mpi.sh [hunk 1]
```diff
@@ -23,6 +24,9 @@ fi
 build_with_ucx=""
 if [[ "$target_platform" == linux-* ]]; then
     build_with_ucx="--with-ucx=$PREFIX"
+    # allow-shlib-undefined required for dependencies to link against older sysroot
+    # avoids undefined
+    wrapper_ldflags="${wrapper_ldflags} -Wl,--allow-shlib-undefined"
```
Comment (minrk) [inline recipe/build-mpi.sh:29]: This is actually the change needed to keep downstream packages from running into the glibc version, as verified by the fact that the tests still pass and still run with 3.12, and do not pass the conda-forge $LDFLAGS.

This change should actually fix #143

File: conda-forge.yml [hunk 2]
```diff
@@ -12,6 +12,8 @@ conda_forge_output_validation: true
 github:
   branch_name: main
   tooling_branch_name: main
+os_version:
+  linux_64: cos7
```
Comment (jakirkham) [inline conda-forge.yml:16]: Even though this is already the default in the other cases (and soon will be for `linux-64`), maybe we should note these explicitly in all cases?

```suggestion
  linux_64: cos7
  linux_aarch64: cos7
  linux_ppc64le: cos7
```

We could also then drop these lines from `conda_build_config.yaml` as `os_version` also configures `c_stdlib_version` for us

https://github.com/conda-forge/openmpi-feedstock/blob/99059c0e090c0f8b6f821cfd5e80c0b1c35f62f0/recipe/conda_build_config.yaml#L1-L2

**Comment by minrk** [comment]
I'd really like to land this to unbreak things caused by #158. If folks are still uncomfortable with including the link flag from $LDFLAGS, I can remove it and isolate that to its own PR.

**Comment by minrk** [comment]
I will merge this today unless there are objections, to prevent more conflicts like #162, since all questions raised so far have been addressed. To reiterate, this PR applies a strict subset of flags than the currently published build, it does not add any. It's just that one (the one that's necessary for building in conda-forge default environments) is now called out by name.

**Pull request merged**